core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/workspace/WorkspaceResourceMonitor.scala

```diff
@@ -66,43 +68,51 @@ class WorkspaceMonitorRouter(val config: WorkspaceManagerResourceMonitorConfig,
 
 }
 
-class WorkspaceResourceMonitor(
+case class WorkspaceResourceMonitor(
   jobDao: WorkspaceManagerResourceMonitorRecordDao,
-  jobRunners: List[WorkspaceManagerResourceJobRunner]
+  jobRunners: Map[JobType, WorkspaceManagerResourceJobRunner]
 )(implicit val executionContext: ExecutionContext)
     extends LazyLogging {
 
-  val registeredRunners: Map[JobType, List[WorkspaceManagerResourceJobRunner]] = jobRunners.groupBy(_.jobType)
-
   /**
     * Run all jobs, and wait for them to complete
+    *
     * @return the message containing the number of uncompleted jobs
     */
   def checkJobs(): Future[CheckDone] = for {
     jobs: Seq[WorkspaceManagerResourceMonitorRecord] <- jobDao.selectAll()
     jobResults <- Future.sequence(jobs.map(runJob))
-  } yield CheckDone(jobResults.count(_ == false))
+  } yield CheckDone(jobResults.count(!_.isDone))
 
   /**
     * Runs the job in all job runners configured for the type of the job
     * Deletes the job if all runners have completed successfully
+    *
     * @return true if all runners for the job completed successfully, false if any failed
     */
-  def runJob(job: WorkspaceManagerResourceMonitorRecord): Future[Boolean] = for {
-    jobResults <- Future.sequence(
-      registeredRunners
-        .getOrElse(job.jobType, List())
-        .map(_.run(job).map(_.complete).recover { case t =>
-          logger.error(s"Exception monitoring WSM Job", t)
-          false
-        })
-    )
-  } yield
-    if (jobResults.forall(a => a)) { // true iff all runners for the given job completed successfully
-      jobDao.delete(job) // remove completed jobs
-      true
-    } else {
-      false
-    }
+  def runJob(job: WorkspaceManagerResourceMonitorRecord): Future[JobStatus] =
+    for {
+      status <- jobRunners.getOrElse(job.jobType, NoopJobRunner)(job).recover { case t: Throwable =>
```

Comment:
Where does the "running" actually happen?

Reply:
a runner is now a function. See the parens around job